### ISSUE TITLE
Rule: QualifierObfuscationRisk - Warn about qualifier class references that break with R8/ProGuard

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#34]
## Problem

`@Qualifier(SomeClass::class)` generates `StringQualifier` using the fully qualified class name. After R8/ProGuard obfuscation, class names change and qualifier lookups fail silently at runtime.

**Related Koin issues:**
- InsertKoinIO/koin-annotations#328
- InsertKoinIO/koin#2135

## Proposed Rule

**Detection:** Warn when `@Qualifier(value = SomeClass::class)` is used. Suggest using `@Named("explicit-string")` instead, or adding ProGuard keep rules.

**Severity:** Warning
**Category:** annotations

## Example

```kotlin
// Bad - breaks after R8 obfuscation
@Single
@Qualifier(DatabaseIdentifier::class)
class MyRepository

// Good - explicit string is obfuscation-safe
@Single
@Named("database")
class MyRepository
```
